### PR TITLE
wrapped `_popstateHandler` with `run.bind`

### PR DESCRIPTION
**Pull request by bekzod**
I am not sure why it was not wrapped in run, as hashLocation same callback is wrapped already https://github.com/emberjs/ember.js/blob/master/packages/ember-routing/lib/location/hash_location.js#L114

### CODE REVIEW

File: packages/ember-routing/lib/location/history_location.js [hunk 1]
```diff
@@ -208,14 +207,14 @@ export default EmberObject.extend({
...
-    this._popstateHandler = () => {
+    this._popstateHandler = run.bind(this, function() {
       // Ignore initial page load popstate event in Chrome
-      if (!popstateFired) {
-        popstateFired = true;
+      if (!this.__popstateFired) {
+        this.__popstateFired = true;
         if (this.getURL() === this._previousURL) { return; }
       }
       callback(this.getURL());
```
Comment (stefanpenner) [inline packages/ember-routing/lib/location/history_location.js]: is it possible `callback` contains the run? If so, we may need to cleanup that side.

Comment (bekzod) [inline packages/ember-routing/lib/location/history_location.js]: I don't think `callback` contains run it starts here https://github.com/emberjs/ember.js/blob/master/packages/ember-routing/lib/system/router.js#L224 and calls various methods in router

Comment (stefanpenner) [inline packages/ember-routing/lib/location/history_location.js]: I'm surprised this (prior to this PR) doesn't trigger the runloops auto-run assertion, causing peoples tests to fail.

We likely need to figure out why, something smells fishy.

**Review by stefanpenner — CHANGES_REQUESTED** [review]

File: packages/ember-routing/lib/location/history_location.js (hunk 1 above)
Comment (krisselden) [reply packages/ember-routing/lib/location/history_location.js]: tests generally don't navigate with the url right?

Comment (rwjblue) [reply packages/ember-routing/lib/location/history_location.js]: Confirm.

**Review by auvipy — CHANGES_REQUESTED** [review]
can u plz fix the conflicts?

**Comment by locks** [comment]
Closing due to no longer being relevant. Thanks for the work!

**Pull request closed without merging**